File: CCIP/CCIP/UIViewController+addition.m
//
//  UIViewController+addition.m
//  CCIP
//
//  Created by 腹黒い茶 on 2016/08/01.
//  Copyright © 2016年 CPRTeam. All rights reserved.
//

#import "UIViewController+addition.h"

@implementation UIView (addition)

- (CGFloat)topGuideHeight {
    return [((UIViewController *)self.nextResponder) topGuideHeight];
}

- (CGFloat)bottomGuideHeight {
    return [((UIViewController *)self.nextResponder) bottomGuideHeight];
}

@end

@implementation UIViewController (addition)

- (CGFloat)topGuideHeight {
    CGFloat guide = 0.0;
    if (self.navigationController.navigationBar.translucent) {
        if (self.prefersStatusBarHidden == NO) guide += 20;
        if (self.navigationController.navigationBarHidden == NO) guide += self.navigationController.navigationBar.bounds.size.height;
    }
    return guide;
}

- (CGFloat)bottomGuideHeight {
    CGFloat guide = 0.0;
    if (self.tabBarController.tabBar.hidden == NO) guide += self.tabBarController.tabBar.bounds.size.height;
    return guide;
}

@end

File: CCIP/CCIP/Class+addition/UIViewController+addition.swift
//
//  UIViewController+addition.swift
//  OPass
//
//  Created by 腹黒い茶 on 2018/11/4.
//  Copyright © 2018 OPass. All rights reserved.
//

import Foundation
import UIKit

@objc extension UIView {
    func topGuideHeight() -> CGFloat {
        return (self.next as! UIViewController).topGuideHeight();
    }

    func bottomGuideHeight() -> CGFloat {
        return (self.next as! UIViewController).bottomGuideHeight();
    }
}

@objc extension UIViewController {
    func topGuideHeight() -> CGFloat {
        var guide : CGFloat = 0.0;
        if (self.navigationController!.navigationBar.isTranslucent) {
            if (self.prefersStatusBarHidden == false) {
                guide += 20;
            }
            if (self.navigationController!.isNavigationBarHidden == false) {
                guide += self.navigationController!.navigationBar.bounds.size.height;
            }
        }
        return guide;
    }

    func bottomGuideHeight() -> CGFloat {
        var guide : CGFloat = 0.0;
        if (self.tabBarController!.tabBar.isHidden == false) {
            guide += self.tabBarController!.tabBar.bounds.size.height;
        }
        return guide;
    }

    func isVisible() -> Bool {
        return self.isViewLoaded && (self.view!.window != nil);
    }
}
